freeCodeCamp/ui
# feat(tools): enable typecheck on all TypeScript files

Checklist:



- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)



I noticed that if we have a TS error in a test or a storybook file, `pnpm run typecheck` would ignore them.

This PR:
- Updates the TypeScript and Rollup config to enable typecheck on all TypeScript files, while still keeping irrelevant types out of the package bundle.
- Converts Jest config and setup files to TypeScript - This is a change I've been wanting to make. I didn't intend to include it in this PR, but I had to in order to run typecheck on test files (details in the PR comments)
- Adds back `ts-node` - It is unfortunately a buddy of Jest ([ref](https://jestjs.io/docs/configuration))
  - <img width="1024" alt="Screenshot 2024-04-24 at 23 07 03" src="https://github.com/freeCodeCamp/ui/assets/25715018/50dabbba-50a8-424c-8176-59cea9ff95b7">
- Removes `tsx`

## Testing

- Make a TS error in a `*.test.tsx` or a `*.storybook.tsx` file
- Run `pnpm run typecheck` and check if it throws an error
- Run `pnpm run build` and check if types in test and storybook files are included in the bundle
